### PR TITLE
UGRID-4/UGRID-8/UGRID-12: Docker Linux build

### PR DESCRIPTION
**Pull request by ahmad-el-sayed**

### CODE REVIEW

File: scripts/module_load.sh [hunk 1]
```diff
@@ -4,5 +4,7 @@
 module --verbose load gcc/12.2.0_gcc12.2.0
```
Comment (lucacarniato) [inline scripts/module_load.sh:4]: do we still need module_load.sh if we build all in docker?

File: tests/api/ApiTests.cpp [hunk 2]
```diff
@@ -12,10 +12,10 @@ static void define_variable_attributes(int file_id, std::string const& variable_
     auto error_code = ugridapi::ug_name_get_long_length(name_long_length);
     ASSERT_EQ(ugridapi::UGridioApiErrors::Success, error_code);
 
-    std::unique_ptr<char> const variable_name_ptr(new char[name_long_length]);
+    std::unique_ptr<char[]> const variable_name_ptr(new char[name_long_length]);
```
Comment (lucacarniato) [inline tests/api/ApiTests.cpp:15]: i propose we write a task to remove operator new in the unit tests as much as possible (using strings, vectors)

**Review by lucacarniato — APPROVED** [review]

File: tests/CMakeLists.txt [hunk 3]
```diff
@@ -1,9 +1,16 @@
 # Download google test
-FetchContent_Declare(
-  googletest
-  GIT_REPOSITORY https://github.com/google/googletest.git
-  GIT_TAG v1.15.0)
-FetchContent_MakeAvailable(googletest)
+# FetchContent_Declare(
```
Comment (lucacarniato) [inline tests/CMakeLists.txt:2]: remove if not needed

**Review by lucacarniato — APPROVED** [review]

File: scripts/module_load.sh (hunk 1 above)
Comment (ahmad-el-sayed) [reply scripts/module_load.sh:4]: no, will remove

File: tests/api/ApiTests.cpp (hunk 2 above)
Comment (ahmad-el-sayed) [reply tests/api/ApiTests.cpp:15]: done

File: scripts/module_load.sh (hunk 1 above)
Comment (ahmad-el-sayed) [reply scripts/module_load.sh:4]: actually, I'll keep them until the sonar cloud job (failed and uses them) is move to docker

File: scripts/docker/install_netcdf_cxx4.sh [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+      IFS=\|
+      echo "${valid_build_types[*]}"
+    )"
+    exit 1
+  fi
+}
+
+# Creates work directory
+function create_directories() {
+  # if [ -d "${WORK_DIR}" ]; then
```
[the remark this answers is not in the document]
Comment (ahmad-el-sayed) [reply scripts/docker/install_netcdf_cxx4.sh:85]: this allows clean builds and installation, which is desired in some situations

File: scripts/docker/install_netcdf_cxx4.sh [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+}
+
+function validate_args() {
+  # Check if all mandatory arguments are provided
+  if [ -z "${WORK_DIR}" ] || [ -z "${INSTALL_DIR}" ]; then
+    usage
+    exit 1
+  fi
+
+  # if provided the release type can be one of Release, Debug, orRelWithDebInfo
```
[the remark this answers is not in the document]
Comment (ahmad-el-sayed) [reply scripts/docker/install_netcdf_cxx4.sh:63]: typo fixed

File: CMakeLists.txt [hunk 6]
```diff
@@ -75,17 +94,25 @@ include_directories(${Boost_INCLUDE_DIR})
...
+    else()
+    message(FATAL_ERROR "Could not find NetCDFCxx" )
 endif()
 
-# FetchContent added in CMake 3.11, downloads during the configure step
-include(FetchContent)
+find_package(hdf5 REQUIRED)
+if (hdf5_FOUND)
+    message(STATUS "Found HDF5 ${hdf5_VERSION}")
+# else()
```
[the remark this answers is not in the document]
Comment (ahmad-el-sayed) [reply CMakeLists.txt:113]: removed

**Pull request merged**